drafts/draft-0005.md

```diff
@@ -0,0 +1,188 @@
+}
+```
+
+##### Response
+
+On success, the endpoint should return a 200 HTTP status code and a JSON-encoded object.
+
+Response fields:
+
+Name | Type | Description
+-----|------|------------
+`id` | string | Required. An identifier for the multisig request that is generated by the multisig service.
+`statusHref` | string | Required. A fully qualified url where an agent can poll for updates on the status of the multisig request. This is specified in the Status Endpoint section.
+`extras` | object | Optional. An object that contains additional application specific response data which could be used to provide an enhanced user experience. For instance, showing a link to authorize a transaction.
+
+Example:
+
+```json
+{
+  "id": "c585ceee-b1b9-4009-aa37-b8544346a036",
+  "statusHref": "https://stellarguard.me/api/transactions/c585ceee-b1b9-4009-aa37-b8544346a036",
+  "extras": {
+    "isStellarGuard": true,
+    "url": "https://stellarguard.me/transactions/c585ceee-b1b9-4009-aa37-b8544346a036"
+  }
+}
+```
+
+#### Status Endpoint
+
+This will be used to query for the status of a multisig transaction.
+
+##### Request
+
+`GET MULTISIG_ENDPOINT>/:id`
+
+Request Path Parameters:
+
+Name | Type | Description
+-----|------|------------
+`id` | string | Required. The id used to look up the status, which was returned in the submit transaction response.
+
+Example:
+
+```
+GET https://stellarguard.me/api/transactions/c585ceee-b1b9-4009-aa37-b8544346a036
+```
+
+##### Response
+
+On success, the endpoint should return a `200 OK` HTTP status code and a JSON-encoded object.
+
+Response fields:
+
+Name | Type | Description
+-----|------|------------
+`id` | string | Required. An identifier for the multisig request that is generated by the multisig service.
+`status` | string | Required. An string indicating the status of the multisig request which can have one of the following values:<br><br>- `pending` - The request requires more signatures<br>- `success` - The transaction has been fully signed<br>- `failed` - There was an error with the transaction and it could not be submitted successfully
+`uri` | string | Required. The SEP-0007 URI that was submitted to the service
+`signers`| string[] | Required. An array of public keys that have signed the transaction. 
+
+
+Example:
+
+```json
+{
+  "id": "c585ceee-b1b9-4009-aa37-b8544346a036",
+  "status": "pending",
+  "uri": "web+stellar:tx?xdr=AAAAAHFd0%2BHQV5u%2FY%2FfM3%2BVelUr1IWwSFL8CUDIAUudUdD4MAAAAZAADyI8AAAADAAAAAAAAAAAAAAABAAAAAAAAAAEAAAAAcV3T4dBXm79j98zf5V6VSvUhbBIUvwJQMgBS51R0PgwAAAAAAAAAAACYloAAAAAAAAAAAVR0PgwAAABAEw8ODG0iixkbAHg1aJATAnZS2531PhGauuSvFDad2WxHKzIenUNbc7K5mGiSpe5jvqe19OQCbNFuBjqN11jfBw%3D%3D",
+  "signers": ["GBYV3U7B2BLZXP3D67GN7ZK6SVFPKILMCIKL6ASQGIAFFZ2UOQ7AZB5V"]
+}
+```
+
```

Comment:
Please for god sake just send me back the raw XDR.

Reply:
Fetch the transaction & its signatures to present them to the user.

Reply:
It's still about having the wallet being able to expose the status of the transaction but somehow you don't want it in there so I really don't get what you're trying to do. I'm going to leave it as it is because now that whole thing is nothing but confusing to me.

Reply:
The wallet can expose the status of the transaction [to the user] by displaying the _signers_ from the response.

I don’t see a benefit of exposing **signatures** in addition to the **signers**. These are two different pieces of information and **signers** is more valuable to expose.

Reply:
> Why is it a problem than signatures can be retrieved?

My concern is security.
You can get signatures for one transaction XDR with a fixed set of parameters (today we only have a callback but in the future we could have other fields too). Today you only sign full transactions but in the future we could have partial signing of transactions.
Forcing a service to expose _sensitive_ information such as signatures in the protocol leaves open an attack vector where these signatures can be retrieved and applied to a request with different parameters, which is why I'm hesitant.

I understand your use case for transparency between these services, however, would you want your coordinator to transparently pass data between co-signers even if that data was _sensitive_ such as private keys or passwords (or signatures in this case, which is also _sensitive_)?

To state my claim here more explicitly... I'm saying that **signatures are sensitive information** and should be treated more securely.

--- 
> If the service retains information it's not coordination anymore.

Correct, this is **NOT** coordination, as I've described a number of times earlier. Coordination is one use case and a coordination protocol can expose signatures if it deems that to be an important part of the coordination, that is **out of scope for this SEP**.

--- 

> I'm not willing to tell user a tx is signed without having the app check the signature

The protocol in the SEP is defining that the multisig service should only display a `signer` in the list if it has the `signature` for it. So if the multisig service is following the protocol then wallets can trust that they have the relevant signature. At the end the wallet is only going to work with a multisig service that the **user** has chosen via the data field so the user will trust the result of the `signers` list from the multisig service. This should be a good enough reason to trust the result of the signers list from the multisig service without verification of the actual signature, no?

Reply:
Look Nikhil you keep saying this is not the coordination SEP but status fetching is clearly a coordination component and must be handled as such, because obviously coordination will depends on this component. If it doesn't, it means you're going to introduce another endpoint that mostly does the same thing than this one and that wouldn't be so great.

Now you got a good point saying that prior to publication signatures can be sensitive, but that's not right to put it on the same level than password or private keys. On the ledger signatures are a public data - password and private keys are not. Please don't add to my confusion with this kind of comparisons.

In a trustless design a signature that is sent to the network must be considered compromised - that is public. Now that's worth thinking twice about this but I'd say that the normal coordinator behavior should be to expose signatures (actually on-chain signature collection is plainly public and trustless and in itself that's not an issue).

There may be situations were signatures must kept secret before publication but I think that would be the symptom of a missing or a wrongly designed feature. If you're planning on offer support for that kind of thing it means you're stepping out of the trustless environment to go back to the trust-a-third party one. This is not the way.

> At the end the wallet is only going to work with a multisig service that the user has chosen via the data field so the user will trust the result of the signers list from the multisig service. This should be a good enough reason to trust the result of the signers list from the multisig service without verification of the actual signature, no?

Again mate we're in the crypto space and we're playing everyday with concepts such as "redundancy", "distributed", "trustless". However there's a strong tendency to go back to the old centralized thinking mode. Would you say that because I trust an Horizon node I don't need a decentralized ledger? No. A proper signature coordinator for a trustless ledger must be trustless as well, right? Else the whole thing is not trustless anymore. I think that's how that game is meant to be played.

Reply:
Resolved

Reply:
I'm not sure if it is resolved or if you want to discuss this some more @nikhilsaraf

Reply:
I don't necessarily agree with the XDR being returned here but we can consider this resolved for now.